source/OptimaJet.DWKit.StarterApplication/Middleware/ErrorReportingJobFilter.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Linq;
+using Bugsnag;
+using Hangfire.Common;
+using Hangfire.States;
+
+namespace OptimaJet.DWKit.StarterApplication.Middleware
+{
+    // See: https://github.com/HangfireIO/Hangfire/issues/553
+    public class ErrorReportingJobFilter : JobFilterAttribute, IElectStateFilter
+    {
+        protected readonly IClient bugsnagClient;
+
+        public ErrorReportingJobFilter(IClient bugsnagClient)
+        {
+            this.bugsnagClient = bugsnagClient;
+        }
+
+        public void OnStateElection(ElectStateContext context)
+        {
+            // the way Hangfire works is retrying a job X times (10 by default), so this wont be called directly with a 
+            // failed state sometimes.
+            // To solve this we should look into TraversedStates for a failed state
+
+            var failed = context.CandidateState as FailedState ??
+                         context.TraversedStates.FirstOrDefault(x => x is FailedState) as FailedState;
+
+            if (failed == null)
```

Comment:
so, this is checking if the job has previously failed, do not report to bugsnag?

Reply:
There are many different state transitions that can initiate this callback.  If this was not a failed condition, then we don't want to notify bugsnag.